dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -198,13 +198,14 @@ public async IAsyncEnumerable<string> UpsertBatchAsync(string collectionName, IE
 
         if (searchResult == null) { yield break; }
 
+        var minAzureSearchScore = CosineSimilarityToScore(minRelevanceScore);
         await foreach (SearchResult<AzureCognitiveSearchMemoryRecord>? doc in searchResult.Value.GetResultsAsync())
         {
-            if (doc == null || doc.Score < minRelevanceScore) { continue; }
+            if (doc == null || doc.Score < minAzureSearchScore) { continue; }
 
             MemoryRecord memoryRecord = doc.Document.ToMemoryRecord(withEmbeddings);
 
-            yield return (memoryRecord, doc.Score ?? 0);
+            yield return (memoryRecord, ScoreToCosineSimilarity(doc.Score ?? 0.333333));
```

Comment:
Maybe have a constant like MaxAcsDistanceScore = 1 / 3; Since this is re-used below too.

Reply:
pls see latest code, I logic is in the function